src/Logger.php

```diff
@@ -129,26 +117,22 @@ public function getTarget($name): ?Target
     }
 
     /**
-     * @param array|Target[] $targets the log targets. Each array element represents a single [[Target|log target]] instance
+     * @param Target[] $targets the log targets. Each array element represents a single [[Target|log target]] instance
      * or the configuration for creating the log target instance.
      */
     public function setTargets(array $targets): void
```

Comment:
Since I cannot typehint as `Target[]` in PHP, should I manually check the passed parameters?

Reply:
Either that or we can implement something like:

```php

class Targets implements \IteratorAggregate
{
    /**
     * @var Target[]
     */
    private $targets = [];

    public function add(Target $target): self
    {
        $new = clone $this;
        $new->targets[] = $target;
        return $new;
    }

    public function getIterator()
    {
        return new \ArrayIterator($this->targets);
    }
}
```

Reply:
> Since I cannot typehint as `Target[]` in PHP

We could also implement this as `setTargets(Target ...$targets)`.

Reply:
Thank you both. Took the simple path for now, will revise if we standardize instances arrays.